python/idsse_common/idsse/common/publish_confirm.py

```diff
@@ -169,14 +170,14 @@ def on_exchange_declareok(self, _unused_frame, userdata):
         logger.info('Exchange declared: %s', userdata)
         self.setup_queue(self._queue)
 
-    def setup_queue(self, queue_name):
+    def setup_queue(self, queue: Queue):
         """Setup the queue on RabbitMQ by invoking the Queue.Declare RPC
         command. When it is complete, the on_queue_declareok method will
         be invoked by pika.
         :param str|unicode queue_name: The name of the queue to declare.
         """
-        logger.info('Declaring queue %s', queue_name)
-        self._channel.queue_declare(queue=queue_name,
+        logger.info('Declaring queue %s', queue.name)
+        self._channel.queue_declare(queue=queue.name,
```

Comment:
Is it simple enough to pass the other queue attributes through to the kwargs of `queue_declare` here, since it accepts optional args like "durable" and "exclusive". 

This would successfully make #14 redundant.

Reply:
```suggestion
        self._channel.queue_declare(queue=queue.name,
                                    durable=queue.durable,
                                    exclusive=queue.exclusive,
                                    auto_delete=queue.auto_delete,
```